pkg/controllermanager/controller/seed/seed.go

```diff
@@ -86,20 +89,26 @@ func NewSeedController(
 	if err != nil {
 		return nil, fmt.Errorf("failed to get Seed Informer: %w", err)
 	}
+	controllerInstallationInformer, err := gardenCache.GetInformer(ctx, &gardencorev1beta1.ControllerInstallation{})
+	if err != nil {
+		return nil, fmt.Errorf("failed to get ControllerInstallation Informer: %w", err)
+	}
 
 	seedController := &Controller{
 		gardenClient: gardenClient,
 		config:       config,
 		log:          log,
 
-		secretsReconciler:    NewSecretsReconciler(gardenClient),
-		lifeCycleReconciler:  NewLifecycleReconciler(gardenClient, config),
-		seedBackupReconciler: NewBackupBucketReconciler(gardenClient),
+		secretsReconciler:        NewSecretsReconciler(gardenClient),
+		lifeCycleReconciler:      NewLifecycleReconciler(gardenClient, config),
+		seedBackupReconciler:     NewBackupBucketReconciler(gardenClient),
+		extensionCheckReconciler: NewExtensionCheckReconciler(gardenClient, metav1.Now),
 
-		secretsQueue:          workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "Seed Secrets"),
-		seedBackupBucketQueue: workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "Backup Bucket"),
-		seedLifecycleQueue:    workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "Seed Lifecycle"),
-		workerCh:              make(chan int),
+		secretsQueue:            workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "Seed Secrets"),
+		seedBackupBucketQueue:   workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "Backup Bucket"),
+		seedLifecycleQueue:      workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "Seed Lifecycle"),
+		seedExtensionCheckQueue: workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "seed-extension-check"),
```

Comment:
Nit: For symmetry reasons:

```suggestion
		seedExtensionCheckQueue: workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "Extension Check"),
```